components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java

```diff
@@ -431,10 +438,40 @@ public static APIDTO fromAPItoDTO(ApiTypeWrapper model, String tenantDomain) thr
         } else {
             apidto = fromAPItoDTO(model.getApi(), tenantDomain);
         }
-        apidto.setEndpointURLs(fromAPIRevisionListToEndpointsList(apidto, tenantDomain));
+
+        if (!apidto.isIsAWSAPI()) {
+            apidto.setEndpointURLs(fromAPIRevisionListToEndpointsList(apidto, tenantDomain));
+        } else {
+            //getting the server url from the swagger to be displayed as the endpoint url in the dev portal for aws apis
+            apidto.setEndpointURLs(setEndpointURLsForAwsAPIs(model, tenantDomain));
+        }
         return apidto;
     }
 
+    private static List<APIEndpointURLsDTO>  setEndpointURLsForAwsAPIs(ApiTypeWrapper model, String tenantDomain) throws APIManagementException {
```

Comment:
```suggestion
    private static List<APIEndpointURLsDTO>  setEndpointURLsForAdvertisedAPIs(ApiTypeWrapper model, String tenantDomain) throws APIManagementException {
```

Reply:
This only works for AWS APIs aiya. This does not work for all advertised APIs. Thats why we have a separate bool to see if its a AWS API.